modules/pulse/pulse.timing.ini

```diff
@@ -3,259 +3,322 @@ description: Timing diagrams for PULSE block
+35      : TRIG=0        -> OUT=1
+37      :               -> QUEUED=1
+40      : TRIG=1        -> 
+41      :               -> QUEUED=0, OUT=0
+45      : TRIG=0        -> OUT=1
+47      :               -> QUEUED=1
+50      : TRIG=1        -> 
+51      :               -> QUEUED=0, OUT=0
+55      : TRIG=0        -> OUT = 1
+57      :               -> QUEUED=1
+61      :               -> QUEUED=0, OUT=0
+65      : ENABLE=0      ->
+
+
+[10 Pulse stretching with no delay activate on falling edge]
+1       : WIDTH_L=5     ->
+2       : DELAY_L=0     ->
+3       : TRIG_EDGE=1   ->
+5       : ENABLE=1      ->
+7       : TRIG=1        ->
+10      : TRIG=0        -> OUT=1
+12      :               -> QUEUED=1
+15      :               -> QUEUED=0
+16      :               -> OUT=0
+20      : ENABLE=0      ->
+
+
+[11 Pulse stretching with no delay activate on both edges]
+1       : WIDTH_L=5     ->
+2       : DELAY_L=0     ->
+3       : TRIG_EDGE=2   ->
+5       : ENABLE=1      ->
 7       : TRIG=1        -> OUT=1
 9       :               -> QUEUED=1
-12      :               -> QUEUED=0, OUT=0
+12      :               -> QUEUED=0
+13      :               -> OUT=0
 21      : TRIG=0        -> OUT=1
 23      :               -> QUEUED=1
-26      :               -> QUEUED=0, OUT=0
-
-[Stretched and delayed pulses too close together]
-1       : WIDTH_L=5
-2       : DELAY_L=10
-5       : ENABLE=1
-14      : TRIG=1
-15      :               -> QUEUED=1
-16      : TRIG=0        -> QUEUED=2
-20      : TRIG=1        -> DROPPED=1
-23      : TRIG=0
-24      :               -> QUEUED=1, OUT=1
-27      : TRIG=1
-28      :               -> QUEUED=2
-29      :               -> QUEUED=2, OUT=0
-33      : TRIG=0
-37      :               -> QUEUED=1, OUT=1
-42      :               -> QUEUED=0, OUT=0
-46      : ENABLE=0
-49      : ENABLE=1      -> DROPPED=0
-
-[No pulses if disabled]
-1       : DELAY_L=4
-4       : TRIG=1
-8       : ENABLE=1
-9       : TRIG=0
-12      : TRIG=1
-13      : TRIG=0        -> QUEUED=1
-14      :               -> QUEUED=2
-16      :               -> QUEUED=1, OUT=1
-17      :               -> QUEUED=0, OUT=0
-18      : ENABLE=1, TRIG=1
-19      :               -> QUEUED=1
-20      : TRIG=0
-21      :               -> QUEUED=2
-22      :               -> QUEUED=1, OUT=1
-23      : ENABLE=0      -> QUEUED=0, OUT=0
-
-[Multiple pulses with no delay]
-1       : WIDTH_L=5
-2       : DELAY_L=0
-3       : STEP_L=9
-4       : PULSES=3
-5       : ENABLE=1
+26      :               -> QUEUED=0
+27      :               -> OUT=0
+30      : ENABLE=0      ->
+
+
+[12 Stretched and delayed pulses too close together]
+1       : WIDTH_L=5     ->
+2       : DELAY_L=12    ->
+5       : ENABLE=1      ->
+14      : TRIG=1        ->
+16      : TRIG=0        -> QUEUED=1
+18      : TRIG=1        ->
+20      :               -> QUEUED=2
+21      : TRIG=0        ->
+25      :               -> QUEUED=1, OUT=1
+27      : TRIG=1        ->
+28      :               -> DROPPED=1
```

Comment:
It happens a clock tick later because one process intercepts the change in state and the following clock tick an evaluation is done by a separate process to which either accepts or rejects the trigger in a number of ways.